src/Schema/OracleSchemaManager.php

```diff
@@ -320,19 +323,226 @@ private function getQuotedIdentifierName($identifier)
+     */
+    private function getDatabaseIndexes(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT IND_COL.TABLE_NAME,
+                 IND_COL.INDEX_NAME AS NAME,
+                 IND.INDEX_TYPE AS TYPE,
+                 DECODE(IND.UNIQUENESS, 'NONUNIQUE', 0, 'UNIQUE', 1) AS IS_UNIQUE,
+                 IND_COL.COLUMN_NAME AS COLUMN_NAME,
+                 IND_COL.COLUMN_POSITION AS COLUMN_POS,
+                 CON.CONSTRAINT_TYPE AS IS_PRIMARY
+            FROM ALL_IND_COLUMNS IND_COL
+       LEFT JOIN ALL_INDEXES IND ON IND.OWNER = IND_COL.INDEX_OWNER AND IND.INDEX_NAME = IND_COL.INDEX_NAME
+       LEFT JOIN ALL_CONSTRAINTS CON ON  CON.OWNER = IND_COL.INDEX_OWNER AND CON.INDEX_NAME = IND_COL.INDEX_NAME
+SQL;
+
+        $conditions = [];
+        $params     = [];
+
+        $conditions[]    = 'IND_COL.INDEX_OWNER = :OWNER';
+        $params['OWNER'] = $databaseName;
+
+        if ($tableName !== null) {
+            $conditions[]         = 'IND_COL.TABLE_NAME = :TABLE_NAME';
+            $params['TABLE_NAME'] = $tableName;
+        }
+
+        $sql .= ' WHERE ' . implode(' AND ', $conditions);
+        $sql .= <<<'SQL'
+        ORDER BY IND_COL.TABLE_NAME, IND_COL.INDEX_NAME, IND_COL.COLUMN_POSITION
+SQL;
+
+        return $this->_conn->fetchAllAssociative($sql, $params);
+    }
+
+    /**
+     * Gets all foreign keys definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    private function getDatabaseForeignKeys(string $databaseName, ?string $tableName = null): array
+    {
+        $sql = <<<'SQL'
+          SELECT COLS.TABLE_NAME,
+                 ALC.CONSTRAINT_NAME,
+                 ALC.DELETE_RULE,
+                 COLS.COLUMN_NAME LOCAL_COLUMN,
+                 COLS.POSITION,
+                 R_COLS.TABLE_NAME REFERENCES_TABLE,
+                 R_COLS.COLUMN_NAME FOREIGN_COLUMN
+            FROM ALL_CONS_COLUMNS COLS
+       LEFT JOIN ALL_CONSTRAINTS ALC ON ALC.OWNER = COLS.OWNER AND ALC.CONSTRAINT_NAME = COLS.CONSTRAINT_NAME
+       LEFT JOIN ALL_CONS_COLUMNS R_COLS ON R_COLS.OWNER = ALC.R_OWNER AND
+                 R_COLS.CONSTRAINT_NAME = ALC.R_CONSTRAINT_NAME AND
+                 R_COLS.POSITION = COLS.POSITION
+SQL;
+
+        $conditions = [];
+        $params     = [];
+
+        $conditions[]    = 'ALC.CONSTRAINT_TYPE = \'R\'';
+        $conditions[]    = 'COLS.OWNER = :OWNER';
+        $params['OWNER'] = $databaseName;
+
+        if ($tableName !== null) {
+            $conditions[]         = 'COLS.TABLE_NAME = :TABLE_NAME';
+            $params['TABLE_NAME'] = $tableName;
+        }
+
+        $sql .= ' WHERE ' . implode(' AND ', $conditions);
+        $sql .= <<<'SQL'
+        ORDER BY COLS.TABLE_NAME, COLS.CONSTRAINT_NAME, COLS.POSITION
+SQL;
+
+        return $this->_conn->fetchAllAssociative($sql, $params);
+    }
+
+    /**
+     * Remove quotes from the given identifier, if needed.
```

Comment:
When is it needed? This code should interact with the database introspection schema of the database which doesn't add any quotes AFAIK. Does it?

Reply:
`listTableDetails` is public API, not internal to schema introspection, and passes `$table`. This parameter may be quoted to reflect lowercase named objects, so its unquoted version need to be used in the queries. If not quoted, then table name must be uppercased (and used like that in queries) to reflect Oracle's strategy.

Reply:
So it's not only about un-quoting but about upper-casing the non-quoted ones. Makes sense but let's name it `normalizeIdentifier()` to match the same method in the platform class.